tests/scripts/hpo/hpo_api_tests.sh

```diff
@@ -148,25 +148,64 @@ function post_experiment_json() {
 	echo "http_code is $http_code Response is ${response}"
 }
 
+# Post a JSON object to HPO(Hyper Parameter Optimization) module
+# input: JSON object
+# output: Create the Curl command with given JSON and get the result
+function stop_experiment() {
+	exp_name=$1
+	echo ""
+	echo "******************************************"
+	echo "stop experiment = ${exp_name}"
+	echo "******************************************"
+
+	form_hpo_api_url "experiment_trials"
+
+	remove_experiment='{"experiment_name":"'${exp_name}'","operation":"EXP_STOP"}'
+
+#	post_cmd=$(curl -s -H 'Content-Type: application/json' ${hpo_url}  -d "${json_array_}"  -w '\n%{http_code}' 2>&1)
```

Comment:
I think this needs to be uncommented and json_array needs to be replaced with remove_experiment

Reply:
That line should be removed, if you look at line #169, i define;
`stop_experiment_cmd="curl -s -H 'Content-Type: application/json' ${hpo_url} -d '${remove_experiment}'  -w '\n%{http_code}'"`

which i use to stop the experiment

Reply:
wrt the error stack trace, I raise an exception to stop the execution of the experiment. This is bubbling all the way up the stack, but I should have caught that exception and handled it gracefully. Note the first WARNING is logged by the optuna framework so we will still see the first stack trace, but I think that is a valid warning, i.e. a trial failed because the experiment was stopped;
```
2022-06-13 23:18:58 - WARNING - optuna.study._optimize - Trial 0 failed because of the following error: Exception('Stopping experiment: petclinic-sample-2-75884c5549-npvgd')
Traceback (most recent call last):
  File "/home/csubrama/.local/lib/python3.8/site-packages/optuna/study/_optimize.py", line 213, in _run_trial
    value_or_values = func(trial)
  File "/home/csubrama/github/john/hpo/src/bayes_optuna/optuna_hpo.py", line 256, in __call__
    actual_slo_value, experiment_status = self.experiment.perform_experiment()
  File "/home/csubrama/github/john/hpo/src/bayes_optuna/optuna_hpo.py", line 110, in perform_experiment
    raise Exception("Stopping experiment: {}".format(self.experiment_name))
```

Reply:
@chandrams I have pushed a change to catch the exception, the server log now look like the following;

```
2022-06-13 20:40:42 - INFO - rest_service - Access server at http://localhost:8085
2022-06-13 20:40:42 - INFO - grpc_service - Starting gRPC server at http://0.0.0.0:50051
2022-06-13 20:40:57 - WARNING - optuna.study._optimize - Trial 0 failed because of the following error: Exception('Stopping experiment: petclinic-sample-2-75884c5549-npvgd')
Traceback (most recent call last):
  File "/home/johara/.local/lib/python3.10/site-packages/optuna/study/_optimize.py", line 213, in _run_trial
    value_or_values = func(trial)
  File "/working/projects/redHat/autotune/hpo/src/bayes_optuna/optuna_hpo.py", line 258, in __call__
    actual_slo_value, experiment_status = self.experiment.perform_experiment()
  File "/working/projects/redHat/autotune/hpo/src/bayes_optuna/optuna_hpo.py", line 110, in perform_experiment
    raise Exception("Stopping experiment: {}".format(self.experiment_name))
Exception: Stopping experiment: petclinic-sample-2-75884c5549-npvgd
2022-06-13 20:40:57 - WARNING - bayes_optuna.optuna_hpo - Experiment stopped: petclinic-sample-2-75884c5549-npvgd
```

Reply:
> That line should be removed, if you look at line #169, i define; `stop_experiment_cmd="curl -s -H 'Content-Type: application/json' ${hpo_url} -d '${remove_experiment}' -w '\n%{http_code}'"`
> 
> which i use to stop the experiment

Sorry, I think my comment was not clear. Above curl command doesn't get executed, we need to enclose it inside $(curl...)

Reply:
Sry, my mistake. I will update the `stop_experiment` function so that `stop_experiment_cmd` is called

Reply:
I have updated the `stop_experiment` function